apps/src/panels/PanelsView.tsx

```diff
@@ -68,96 +72,144 @@ const PanelsView: React.FunctionComponent<PanelsProps> = ({
   }, [targetWidth, targetHeight]);
 
   const handleButtonClick = useCallback(() => {
-    if (currentPanel < panels.length - 1) {
-      setCurrentPanel(currentPanel + 1);
+    if (currentPanelIndex < panels.length - 1) {
+      setCurrentPanelIndex(currentPanelIndex + 1);
     } else {
-      onContinue(panels[currentPanel].nextUrl);
+      onContinue(panels[currentPanelIndex].nextUrl);
     }
-  }, [panels, currentPanel, onContinue]);
+  }, [panels, currentPanelIndex, onContinue]);
 
   const handleBubbleClick = (index: number) => {
-    setCurrentPanel(index);
+    setCurrentPanelIndex(index);
   };
 
   // Reset to first panel whenever panels content changes if specified.
   useEffect(() => {
     if (resetOnChange) {
-      setCurrentPanel(0);
+      setCurrentPanelIndex(0);
     }
   }, [panels, resetOnChange]);
 
-  // Reset to last panel if number of panels has reduced
+  // Reset to last panel if number of panels has reduced.
   useEffect(() => {
-    if (currentPanel >= panels.length) {
-      setCurrentPanel(Math.max(panels.length - 1, 0));
+    if (currentPanelIndex >= panels.length) {
+      setCurrentPanelIndex(Math.max(panels.length - 1, 0));
     }
-  }, [currentPanel, panels]);
+  }, [currentPanelIndex, panels]);
 
   // Cancel any in-progress text-to-speech when the panel changes.
   useEffect(() => {
     if (offerTts) {
       cancelSpeech();
     }
-  }, [currentPanel, offerTts]);
+  }, [currentPanelIndex, offerTts]);
 
-  const panel = panels[currentPanel];
+  // Reset typing if the panel changes.
+  useEffect(() => {
+    setTypingDone(false);
+  }, [currentPanelIndex, setTypingDone]);
+
+  const panel = panels[currentPanelIndex];
   if (!panel) {
     return null;
   }
 
-  const showSmallText = height < 300;
+  const previousPanel =
+    panel.fadeInOverPrevious &&
+    currentPanelIndex > 0 &&
+    panels[currentPanelIndex - 1]
+      ? panels[currentPanelIndex - 1]
+      : null;
 
   const layoutClassMap = {
-    'text-top-left': styles.markdownTextTopLeft,
-    'text-top-center': styles.markdownTextTopCenter,
-    'text-bottom-left': styles.markdownTextBottomLeft,
-    'text-bottom-center': styles.markdownTextBottomCenter,
-    'text-bottom-right': styles.markdownTextBottomRight,
-    'text-top-right': styles.markdownTextTopRight,
+    'text-top-left': styles.textTopLeft,
+    'text-top-center': styles.textTopCenter,
+    'text-bottom-left': styles.textBottomLeft,
+    'text-bottom-center': styles.textBottomCenter,
+    'text-bottom-right': styles.textBottomRight,
+    'text-top-right': styles.textTopRight,
   };
 
   const textLayoutClass = panel.layout
     ? layoutClassMap[panel.layout]
-    : styles.markdownTextTopRight;
+    : styles.textTopRight;
+
+  const buttonText =
+    currentPanelIndex < panels.length - 1
+      ? commonI18n.next()
+      : commonI18n.continue();
+
+  const plainText = markdownToTxt(panel.text);
```

Comment:
Does this mean that the typing effect doesn't support markdown? Probably something to call out/validate for levelbuilders if we end up adding this to the level edit page?

Reply:
That's what I found so far.  It would be neat, though, so we should investigate further at some point.